diff --git a/hls/toe/tx_app_stream_if/tx_app_stream_if.cpp b/hls/toe/tx_app_stream_if/tx_app_stream_if.cpp
index c0afdf3..51bc190 100755
--- a/hls/toe/tx_app_stream_if/tx_app_stream_if.cpp
+++ b/hls/toe/tx_app_stream_if/tx_app_stream_if.cpp
@@ -106,7 +106,7 @@ void tasi_metaLoader(	stream<appTxMeta>&			appTxDataReqMetaData,
 			{
 				// TODO there seems some redundancy
 				ap_uint<32> pkgAddr;
-				pkgAddr(31, 30) = 0x01;
+				pkgAddr(31, 30) = 0x00;
 				pkgAddr(29, WINDOW_BITS) = tasi_writeMeta.sessionID(13, 0);
 				pkgAddr(WINDOW_BITS-1, 0) = writeSar.mempt;
 
diff --git a/hls/toe/tx_engine/tx_engine.cpp b/hls/toe/tx_engine/tx_engine.cpp
index c5ab022..8bff620 100755
--- a/hls/toe/tx_engine/tx_engine.cpp
+++ b/hls/toe/tx_engine/tx_engine.cpp
@@ -217,7 +217,7 @@ void metaLoader(hls::stream<extendedEvent>&				eventEng2txEng_event,
 				currLength = (txSar.app - ((ap_uint<WINDOW_BITS>)txSar.not_ackd));
 				// Construct address before modifying txSar.not_ackd
 				ap_uint<32> pkgAddr;
-				pkgAddr(31, 30) = 0x01;
+				pkgAddr(31, 30) = 0x00;
 				pkgAddr(29, WINDOW_BITS) = ml_curEvent.sessionID(13, 0);
 				pkgAddr(WINDOW_BITS-1, 0) = txSar.not_ackd(WINDOW_BITS-1, 0); //ml_curEvent.address;
 
@@ -333,7 +333,7 @@ void metaLoader(hls::stream<extendedEvent>&				eventEng2txEng_event,
 
 				// Construct address before modifying txSar.ackd
 				ap_uint<32> pkgAddr;
-				pkgAddr(31, 30) = 0x01;
+				pkgAddr(31, 30) = 0x00;
 				pkgAddr(29, WINDOW_BITS) = ml_curEvent.sessionID(13, 0);
 				pkgAddr(WINDOW_BITS-1, 0) = txSar.ackd(WINDOW_BITS-1, 0); //ml_curEvent.address;
 
